doc/operation-and-maintenance/Reloading-configuration-on-a-running-system.md

```diff
@@ -22,3 +27,147 @@ The following options' changes will be ignored when using `mongooseimctl` tool:
 * s2s_*
 * all_metrics_are_global
 * odbc_*
+
+
+### Node-specific options
+
+Very rarely we want different configs for each node in cluster.
+Than `reload_cluster` would detect configuration inconsistency and would not
+allow configuration updates.
+
+To tell `reload_cluster` to ignore such options, extra information should be
+provided in `ejabberd.cfg`.
+
+It's called `node_specific_options`.
+
+They are defined on top level of the configuration file using
+`node_specific_options` tuple. This tuple should be the same for all configs
+in cluster.
+
+`node_specific_options` contains a list of match patterns. If you are familiar
+with ETS tables or Mnesia tuple matching - it's the same thing.
+Match patterns are documented in
+[Erlang/OTP docs](http://erlang.org/doc/apps/erts/match_spec.html).
+
+The pattern mechanism is also documented in
+[Learn you some Erlang book](http://learnyousomeerlang.com/ets).
+
+Basically, it allows you to put `'_'` to match every possible host.
+
+`[h,'_',module_opt,mod_muc_log, outdir]` would match
+`[h,<<"localhost">>,module_opt,mod_muc_log, outdir]` and
+`[h,<<"any.other.host">>,module_opt,mod_muc_log, outdir]`.
+
+Example showing where to put `node_specific_options`.
+
+```erlang
+{hosts, ["localhost"]}.
+{node_specific_options, [
+    [h,'_',module_opt,mod_muc_log, outdir]
+ ]}.
+{modules, [....]}.
+```
+
+The `node_specific_options` patterns are matched against flat configuration
+options. To print your config in flat form, use the command with running
```

Comment:
To print your config in a flat form, use the command with a running node `mongooseimctl print_flat_config`. ?

To print your config in a form of flat, use the command with a running node `mongooseimctl print_flat_config`. ?

To print your config in a form of flat, use the command while running the `mongooseimctl print_flat_config`node. ?

Reply:
"flat form" is a method of rewriting.
Actually, I often see similar cases it's used without "a".

https://revisionmaths.com/gcse-maths-revision/number/standard-form
https://en.wikipedia.org/wiki/Canonical_form